spec/rubocop/cop/style/slicing_with_range_spec.rb

```diff
@@ -141,5 +187,97 @@
         ary[..-1]
       RUBY
     end
+
+    context 'when calling `[]` with a dot' do
+      it 'reports an offense and corrects for `nil..42`' do
+        expect_offense(<<~RUBY)
+          ary.[](nil..42)
+             ^^^^^^^^^^^^ Prefer `..42` over `nil..42`.
+        RUBY
+
+        expect_correction(<<~RUBY)
+          ary.[](..42)
+        RUBY
+      end
+
+      it 'reports an offense and corrects for `nil..42` without parens' do
+        expect_offense(<<~RUBY)
+          ary.[] nil..42
+             ^^^^^^^^^^^ Prefer `..42` over `nil..42`.
+        RUBY
+
+        expect_correction(<<~RUBY)
+          ary.[] ..42
+        RUBY
+      end
+    end
+
+    context 'when calling `[]` with a safe navigation' do
+      it 'reports an offense and removes the entire method call for 0..-1' do
+        expect_offense(<<~RUBY)
+          ary&.[](0..-1)
+             ^^^^^^^^^^^ Remove the useless `&.[](0..-1)`.
+        RUBY
+
+        expect_correction(<<~RUBY)
+          ary
+        RUBY
+      end
+
+      it 'reports an offense and removes the entire method call for 0..-1 without parens' do
+        expect_offense(<<~RUBY)
+          ary.[] 0..-1
+             ^^^^^^^^^ Remove the useless `.[] 0..-1`.
+        RUBY
+
+        expect_correction(<<~RUBY)
+          ary
+        RUBY
+      end
+
+      it 'reports an offense and corrects for `1..-1`' do
+        expect_offense(<<~RUBY)
+          ary.[](1..-1)
+             ^^^^^^^^^^ Prefer `1..` over `1..-1`.
+        RUBY
+
+        expect_correction(<<~RUBY)
+          ary.[](1..)
+        RUBY
+      end
+
+      it 'reports an offense and corrects for `1..-1` without parens' do
+        expect_offense(<<~RUBY)
+          ary.[] 1..-1
+             ^^^^^^^^^ Prefer `1..` over `1..-1`.
+        RUBY
+
+        expect_correction(<<~RUBY)
+          ary.[] 1..
+        RUBY
+      end
+
+      it 'reports an offense and corrects for `nil..42`' do
+        expect_offense(<<~RUBY)
+          ary&.[](nil..42)
+             ^^^^^^^^^^^^^ Prefer `..42` over `nil..42`.
+        RUBY
+
+        expect_correction(<<~RUBY)
+          ary&.[](..42)
+        RUBY
+      end
+
+      it 'reports an offense and corrects for `nil..42` without parens' do
+        expect_offense(<<~RUBY)
+          ary&.[] nil..42
+             ^^^^^^^^^^^^ Prefer `..42` over `nil..42`.
+        RUBY
+
+        expect_correction(<<~RUBY)
+          ary&.[] ..42
```

Comment:
When parentheses are omitted, the `nil` at the begin of the range must be preserved; otherwise, the semantic change. An offense should not be registered in this case.

```console
$ ruby-parse -e "ary&.[] nil..42"
(csend
  (send nil :ary) :[]
  (irange
    (nil)
    (int 42)))

$ ruby-parse -e "ary&.[] ..42"
(irange
  (csend
    (send nil :ary) :[])
  (int 42))
```